optuna/storages/_base.py

```diff
@@ -521,7 +523,12 @@ def get_trial(self, trial_id: int) -> FrozenTrial:
         raise NotImplementedError
 
     @abc.abstractmethod
-    def get_all_trials(self, study_id: int, deepcopy: bool = True) -> List[FrozenTrial]:
+    def get_all_trials(
+        self,
+        study_id: int,
+        deepcopy: bool = True,
+        state: Optional[Union[Tuple[TrialState, ...], TrialState]] = None,
```

Comment:
[Question] Do we need to accept `TrialState`? Personally tuple of `TrialState`s is enough.

Reply:
Good point. Curious but do you think we should do that for `Study.get_trials` as well? I tried to make `Study.get_trials` as "user-friendly" as possible and propagated this design down to the storage. But this interface might be an anti-pattern in the first place. I'd go with a `Sequence` and not a `Tuple` to be exact in that case, and name the argument `states`.

Reply:
Elaboration: I believe that users of this new argument peruse our documentation and/or catch up with Optuna well; thus, I think it's acceptable to make `state` to be a container even if users pass only one `TrialState`. By restricting `state` to be a container, developers and others who read Optuna codebase will enjoy some benefits, including easier maintenance and clearer logic.

Reply:
Fixed!